### DISCUSSION TITLE
How to prevent exceptions to be printed twice ?

### DISCUSSION THREAD

**Discussion by vgalisson** [gijzelaerr/python-snap7#330] [Q&A] [answered]
Hi,

When an exception occurs, its message is displayed on stdout (probably directly from the snap7 library?) and then the exception is thrown in Python which result in the message being displayed this like so :

![image](https://user-images.githubusercontent.com/22483199/145435268-4ba011ed-1725-4e76-a343-06fbf143894d.png)

Is there a way to prevent the first message (`b'TCP : Invalid address'`) to be displayed ?
Thanks in advance for any help,

**Comment by swamper123** [comment]
Could you provide the minimal pythoncode you have used?

In general, if stuff is send to stdout inside of snap7, we can't handle it. 
python-snap7 is just a wrapper around snap7 and what we can do is to check the error, which is a result of a snap7 function (which is the thing you see up there, beginning at row 2).

**Reply by vgalisson** [reply]
Here is a minimal example (no need to change the "XXXX")
```
import snap7
s7_client = snap7.client.Client()
connect = s7_client.connect("XXXX", 0, 0, 102)
connect = s7_client.connect("XXXX", 0, 0, 102)
```

**Comment by gijzelaerr** [comment]
The line starts with `b''`, which means it comes from python. My guess is that someone forgot to remove a debug print() statement from the python-snap7 code.

**Comment by gijzelaerr** [comment, marked as answer]
couldn't find a print, so my guess is then it is just the python logging. You could reconfigure the python logging module to set it to level `CRITICAL`, which would not print error messages.

so somewhere in your python code add something like:
```
import logging
logging.setLevel(logging.CRITICAL)
```
(untested)

**Reply by vgalisson** [reply]
Hi, thanks for the reply. It was indeed the logging library. There is no `setLevel` for `logging` directly so I used `basicConfig` instead.
```
import logging
logging.basicConfig(level=logging.CRITICAL)
```
To set only the `snap7` logging to `CRITICAL` and not all log, one might use this:
```
import logging
logger = logging.getLogger("snap7")
logger.setLevel(logging.CRITICAL)
```
